### PR TITLE
refactor: Resource locator review changes

### PR DESCRIPTION
**Pull request by valya**

### CODE REVIEW

File: packages/workflow/src/Expression.ts
Comment (mutdmour) [inline packages/workflow/src/Expression.ts:520]: should this be removed before merging?

File: packages/core/src/ExtractValue.ts
Comment (mutdmour) [inline packages/core/src/ExtractValue.ts:142]: should this be removed for merging? otherwise we should use the logger if possible

File: packages/core/src/ExtractValue.ts
Comment (mutdmour) [inline packages/core/src/ExtractValue.ts:105]: should we remove this?

File: packages/design-system/theme/src/input.scss
Comment (mutdmour) [inline packages/design-system/theme/src/input.scss:219]: conflict resolving issue?

File: packages/editor-ui/src/Interface.ts
Comment (mutdmour) [inline packages/editor-ui/src/Interface.ts:24]: unused imports?

File: packages/editor-ui/src/components/ParameterInput.vue
Comment (mutdmour) [inline packages/editor-ui/src/components/ParameterInput.vue:57]: conflict issue?

File: packages/editor-ui/src/plugins/i18n/locales/en.json
Comment (mutdmour) [inline packages/editor-ui/src/plugins/i18n/locales/en.json:858]: conflict issue?

File: packages/core/src/ExtractValue.ts
Comment (mutdmour) [inline packages/core/src/ExtractValue.ts:108]: this error does not match logic.. would not this be thrown if id regex does not match?

File: packages/core/src/ExtractValue.ts
Comment (mutdmour) [inline packages/core/src/ExtractValue.ts:54]: would this stop execution? if property is set that cannot be found? does it need to?

File: packages/nodes-base/nodes/Airtable/Airtable.node.ts
Comment (mutdmour) [inline packages/nodes-base/nodes/Airtable/Airtable.node.ts:512]: why is this `extractValue` key necessary? could not we simply check the param type? would be simpler to roll out RL without having to update the code everywhere.

Comment (valya) [reply packages/nodes-base/nodes/Airtable/Airtable.node.ts:512]: We don't know what the param type is at this point. Finding out what type something is can be a bit intense depending on how deep it is or based on if it has `displayOptions` or not. It's behind a flag because this could cause performance regressions and this we decided this was a good compromise.

File: packages/editor-ui/src/Interface.ts
Comment (valya) [reply packages/editor-ui/src/Interface.ts:24]: These are actually in use. I don't know how other branches are building without them though.

File: packages/core/src/ExtractValue.ts
Comment (valya) [reply packages/core/src/ExtractValue.ts:55]: It will stop execution. This probably shouldn't happen except when developing a node.

File: packages/workflow/src/Expression.ts
Comment (valya) [reply packages/workflow/src/Expression.ts]: I've changed this to do some type casts. It's happening because the resource locator value is the only properly typed object that gets to that point. This causes some oddities with the types. I just cast them because it shouldn't cause any issues and it produces the expected results.

File: packages/nodes-base/nodes/Airtable/Airtable.node.ts
Comment (mutdmour) [reply packages/nodes-base/nodes/Airtable/Airtable.node.ts:512]: maybe I am lost here, but I followed the code and i am not even sure this is getting used.. we should sync on this.

File: packages/editor-ui/src/Interface.ts
Comment (mutdmour) [reply packages/editor-ui/src/Interface.ts:24]: huh ya, except IResourceLocatorResult is not used

File: packages/core/src/ExtractValue.ts
Comment (mutdmour) [reply packages/core/src/ExtractValue.ts:55]: yes, it's a bit of an edge case.. I would lean towards not breaking execution.. but I will defer to you on this one..

**Review by mutdmour — APPROVED** [review]

**Pull request merged**
